procsim/biomass/level0_product_generator.py

```diff
@@ -205,13 +205,24 @@ def generate_output(self):
         if self._hdr.begin_position is None or self._hdr.end_position is None:
             raise ScenarioError('begin/end position must be set')
 
-        # If not read from an input product, use begin/end position as starting point
+        # If not read from an input product, set validity time to slice bounds.
+        # Get slice bounds from middle of slice to deal with merged slices.
+        middle_position = self._hdr.begin_position + (self._hdr.end_position - self._hdr.begin_position) / 2
+        slice_bounds = self._get_slice_frame_interval(middle_position, constants.SLICE_GRID_SPACING)
         if self._hdr.validity_start is None:
-            self._hdr.validity_start = self._hdr.begin_position
-            self._logger.debug('Use begin_position as input for validity start time')
+            if slice_bounds is None:
+                self._logger.warning(f'Could not find slice bounds for central position {middle_position}. Using {self._hdr.begin_position} as validity start.')
```

Comment:
Do not log internal stuff as warning! begin_position is from what? Scenario? Or input file? I think it must be just 'begin_position' here.

Reply:
I'm not sure what you mean by "it must be just 'begin_position' here", but I've removed the internal time from the logger warning in https://github.com/stcorp/procsim/pull/21/commits/d2350bca36a9669cd66e650ebe333fbd16ef6eae.